### ISSUE TITLE
Pushing package using 3.4.4RTM gives Source parameter was not specified.

### ISSUE THREAD

**Issue by FransBouma** [NuGet/Home#3066]
Probably related to #2288 but I don't use a DefaultPushSource specification. 

```
C:\Myprojects\VS.NET Projects\Github\SolutionsDesign\BCLExtensions\src\bin\Release>nuget push SD.Tools.BCLExtensions.1.1
.0-Pre1.nupkg
Checking for updates from https://www.nuget.org/api/v2/.
Currently running NuGet.exe 3.4.4-rtm-1321.
Updating NuGet.exe to 3.4.4-rtm-final.
Update successful.
Source parameter was not specified.

C:\Myprojects\VS.NET Projects\Github\SolutionsDesign\BCLExtensions\src\bin\Release>nuget push SD.Tools.BCLExtensions.1.1
.0-Pre1.nupkg
Source parameter was not specified.

C:\Myprojects\VS.NET Projects\Github\SolutionsDesign\BCLExtensions\src\bin\Release>nuget source
Registered Sources:

  1.  nuget.org [Enabled]
      https://api.nuget.org/v3/index.json
  2.  Microsoft Visual Studio Offline Packages [Enabled]
      C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\
```

Do I have to restate the API key for every package I have to upload, even though I am the owner of all of them and am the same user? 

Doing `nuget setApiKey <myApiKey>` doesn't fix it, even though it says it will store it for the nuget gallery. 

The NuGet.config file contains proper nuget urls, `nuget push -help` states:

```
 -Source                (src)  Specifies the server URL. If not specified, nuget.org is used unless DefaultPushSource config value is set in the NuGet config file.
```

So, it should default to nuget.org, not require a mandatory -Source flag. 

If I append `-Source nuget.org` it requests the ApiKey which is present already.

I use on the same machine nuget push <package> successfully for a long time now from an msbuild based build script, started using the same windows user, also from CLI. 

Edit: nuget.config:

```
C:\Myprojects\VS.NET Projects\Github\SolutionsDesign\BCLExtensions\src\bin\Release>type %Appdata%\NuGet\nuget.config
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageRestore>
    <add key="enabled" value="True" />
    <add key="automatic" value="True" />
</packageRestore>
  <apikeys>
    <add key="https://www.nuget.org" value="REDACTED" />
    <add key="http://nuget.gw.symbolsource.org/Public/NuGet" value="REDACTED" />
    <add key="https://nuget.smbsrc.net/" value="REDACTED" />
  </apikeys>
  <packageSources>
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
    <add key="nuget.org" value="https://www.nuget.org/api/v2/" />
  </packageSources>
  <activePackageSource>
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
  </activePackageSource>
</configuration>
```

(redacted the encrypted keys for obvious reasons :))

**Comment by harikmenon** [comment]
The API key issue is fixed in https://github.com/NuGet/Home/issues/2627. Source parameter become mandatory a couple of releases back (3.4.3 or the one before). You need to either use the DefaultPushSource key in nuget.config to set it or specify it explicitly in the CLI.

You are correct that we havent updated the help text on nuget.exe for this breaking change, thanks for pointing it out. We will fix it.

**Comment by FransBouma** [comment]
If I have to set DefaultPushSource, what does activePackageSource mean in that context? Why is it required to set DefaultPushSource if there's a source set as active? Or is 'Active' only for pull, not push? very vague...

**Comment by FransBouma** [comment]
Ok, I now get the setApiKey crap. I have no way to push this package automatically to the central, default nuget store now using 3.4.4 RTM?

I've specified the DefaultPushSource in nuget.config. I used several versions, the v2 resulted in 'not allowed', the v3 one:

```
  <config>
     <add key="DefaultPushSource" value="https://api.nuget.org/v3/index.json" protocolVersion="3"/>
  </config>
```

apparently goes through but I now am greeted with this:

```
C:\Myprojects\VS.NET Projects\Github\SolutionsDesign\BCLExtensions\src\bin\Release>nuget push SD.Tools.BCLExtensions.1.1
.0-Pre1.nupkg
WARNING: No API Key was provided and no API Key could be found for 'https://www.nuget.org/api/v2/package'. To save an AP
I Key for a source use the 'setApiKey' command.
Pushing SD.Tools.BCLExtensions.1.1.0-Pre1.nupkg to 'https://www.nuget.org/api/v2/package'...
  PUT https://www.nuget.org/api/v2/package/
  Unauthorized https://www.nuget.org/api/v2/package/ 823ms
Please provide credentials for: https://api.nuget.org/v3/index.json
UserName:
```

Which is unacceptable. What to do? I've added the API key already and this worked OK before. Do I have to specify the API key on the command line till 3.5 RTM is released, whenever that may be?

**Comment by FransBouma** [comment]
Ok setting the apikey for https://www.nuget.org/api/v2/package directly works for 3.4.4:

```
nuget setapikey <your apikey> -Source https://www.nuget.org/api/v2/package
```

It then will push using DefaultPushSource without asking for credentials

**Comment by harikmenon** [comment]
Yes, that is correct. the SetAPIKey issue will be fixed in the next release.Both the v2 and v3 API endpoints work when you use the parameter directly in the command.

**Comment by FransBouma** [comment]
Ok, as the problem has a workaround for 3.4.4 (see above) and real fixes on its way in 3.5, I'll close the issue.

**Comment by harikmenon** [comment]
Thanks!!!
